runtime_builders/publish_runtime.py

```diff
@@ -103,29 +106,34 @@ def _resolve_tag(image):
     Given a path to a tagged Docker image in GCR, replace the tag with its
     corresponding sha256 digest.
     """
-    if image is None:
-        raise Exception('Please provide image')
-
     if ':' not in image:
-        logging.info('Image does not contain tag: defaulting to "latest"')
-        base_image = image
-        target_tag = 'latest'
+        logging.error('Image must contain explicit tag!')
+        sys.exit(1)
     else:
-        base_image = image.split(':')[0]
-        target_tag = image.split(':')[1].strip('${}')
+        m = regex.search(TAG_REGEX, image)
```

Comment:
because I'm using a positive lookbehind when searching for the target string, it actually doesn't work if I try and do a `match` from the beginning of the string. I probably could have formatted this regex better haha.